scripts/devel/build_and_install.py

```diff
@@ -40,10 +42,27 @@ def call_abi_checker(workspace_root, ros_version, env):
     pkg_names = [pkg.name for pkg in pkgs.values()]
     assert pkg_names, 'No packages found in the workspace'
 
+    # Filter packages in source space that has been released
+    index = rosdistro.get_index(rosdistro.get_index_url())
+    dist_file = rosdistro.get_distribution_file(index, env['ROS_DISTRO'])
+
+    pkg_names_released = []
+    for pkg_name in pkg_names:
+        try:
+            # Check in released packages for pkg_name (KeyError if unreleased)
+            dist_file.release_packages[pkg_name]
+            pkg_names_released.append(pkg_name)
+        except KeyError:
+            # skip unreleased packages, nothing to do with ABI checking
+            pass
```

Comment:
Please rewrite the `try` / `except` logic to be more condensed like:

```
if pkg_name in dist_file.release_packages:
    pkg_names_released.append(pkg_name)
```

Reply:
Forget my last comment. The whole loop can be condensed to:

```
pkg_names_released = [pkg_name for pkg_name in pkg_names if pkg_name in dist_file.release_packages]
```

Reply:
I like it e4eee53